src/components/list/list-renderer.tsx

```diff
@@ -189,11 +191,11 @@ export class ListRenderer {
      * @returns {HTMLElement} the icon element
      */
     private renderIcon(config: ListRendererConfig, item: ListItem) {
-        const style = {};
-        if (item.iconColor) {
-            config.badgeIcons
-                ? (style['--icon-background-color'] = item.iconColor)
-                : (style['color'] = item.iconColor); // tslint:disable-line:no-string-literal
+        const style: any = {};
+        if (item.iconColor && config.badgeIcons) {
+            style['--icon-background-color'] = item.iconColor;
+        } else if (item.iconColor && !config.badgeIcons) {
```

Comment:
If `item.iconColor && config.badgeIcons` is falsy and `item.iconColor` is truthy, then we already know that `config.badgeIcons` is falsy 🙂 

```suggestion
        } else if (item.iconColor) {
```

Unless you added it as way to harden against bugs appearing as a result of future code changes?

Reply:
:zap: